internal/agent/agent_syncer.go

```diff
@@ -100,6 +100,9 @@ func (syncer *AgentSyncer) checkConfigUpdate() bool {
 			return false
 		}
 		syncer.configRevision = newRevision
+
+		syncer.numaLogger.SetLevel(syncer.config.LogLevel)
```

Comment:
Why do we need to set log level here again as we already set it in main?

Reply:
I thought to update the log level after every GetConfig call since the level could be changed in the configuration and we want to apply the new level.